### ISSUE TITLE
error using eslint-watch package

### ISSUE THREAD

**Issue by digitalnoiz** [coryhouse/pluralsight-redux-starter#1]
using the `npm run lint` command is encountering an error. 

I'm running on a surface pro 3 under windows 10 pro 64bit(10.0, Build 10586)

Here is the npm-debug.log file that is generated:

0 info it worked if it ends with ok
1 verbose cli [ 'C:\\Program Files\\nodejs\\node.exe',
1 verbose cli   'C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'run-script',
1 verbose cli   'lint' ]
2 info using npm@3.8.9
3 info using node@v6.2.0
4 verbose run-script [ 'prelint', 'lint', 'postlint' ]
5 info lifecycle pluralsight-redux-starter@1.0.0~prelint: pluralsight-redux-starter@1.0.0
6 silly lifecycle pluralsight-redux-starter@1.0.0~prelint: no script for prelint, continuing
7 info lifecycle pluralsight-redux-starter@1.0.0~lint: pluralsight-redux-starter@1.0.0
8 verbose lifecycle pluralsight-redux-starter@1.0.0~lint: unsafe-perm in lifecycle true
9 verbose lifecycle pluralsight-redux-starter@1.0.0~lint: PATH: _path omitted - ds_
10 verbose lifecycle pluralsight-redux-starter@1.0.0~lint: CWD: C:\Users\dave.scheffing\source\actdux_orig
11 silly lifecycle pluralsight-redux-starter@1.0.0~lint: Args: [ '/d /s /c', 'esw webpack.config.* src tools' ]
12 silly lifecycle pluralsight-redux-starter@1.0.0~lint: Returned: code: 1  signal: null
13 info lifecycle pluralsight-redux-starter@1.0.0~lint: Failed to exec lint script
14 verbose stack Error: pluralsight-redux-starter@1.0.0 lint: `esw webpack.config.* src tools`
14 verbose stack Exit status 1
14 verbose stack     at EventEmitter.<anonymous> (C:\Program Files\nodejs\node_modules\npm\lib\utils\lifecycle.js:245:16)
14 verbose stack     at emitTwo (events.js:106:13)
14 verbose stack     at EventEmitter.emit (events.js:191:7)
14 verbose stack     at ChildProcess.<anonymous> (C:\Program Files\nodejs\node_modules\npm\lib\utils\spawn.js:24:14)
14 verbose stack     at emitTwo (events.js:106:13)
14 verbose stack     at ChildProcess.emit (events.js:191:7)
14 verbose stack     at maybeClose (internal/child_process.js:850:16)
14 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:215:5)
15 verbose pkgid pluralsight-redux-starter@1.0.0
16 verbose cwd C:\Users\dave.scheffing\source\actdux_orig
17 error Windows_NT 10.0.10586
18 error argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run-script" "lint"
19 error node v6.2.0
20 error npm  v3.8.9
21 error code ELIFECYCLE
22 error pluralsight-redux-starter@1.0.0 lint: `esw webpack.config.* src tools`
22 error Exit status 1
23 error Failed at the pluralsight-redux-starter@1.0.0 lint script 'esw webpack.config.* src tools'.
23 error Make sure you have the latest version of node.js and npm installed.
23 error If you do, this is most likely a problem with the pluralsight-redux-starter package,
23 error not with npm itself.
23 error Tell the author that this fails on your system:
23 error     esw webpack.config.* src tools
23 error You can get information on how to open an issue for this project with:
23 error     npm bugs pluralsight-redux-starter
23 error Or if that isn't available, you can get their info via:
23 error     npm owner ls pluralsight-redux-starter
23 error There is likely additional logging output above.
24 verbose exit [ 1, true ]

**Comment by coryhouse** [comment]
I just removed the npm-shrinkwrap file from this repo. It was generated on my OSX machine which was likely causing you issues. Please delete your npm-shrinkwrap file, re-run npm install, and report back. After this change, it's working fine for me on Win 10 64 with Node 6.2.0.

Thanks for reporting!

**Comment by digitalnoiz** [comment]
Hi Cory,

Thanks for the update but I'm afraid the error is still occurring if I attempt the run just the lint commands directly. The lint:watch command and full start work fine so I think I can use those to get by. This error appears after the linting is actually done and reported so I see the eslint results then this error. After experimenting with it a little, it seems eslint encounters a problem if you try running it as an npm script. If I run the command directly from the command line `node_modules/.bin/esw webpack.config.* src tools` or `node_modules/.bin/eslint webpack.config.* src tools`, it runs fine. However if I add either of these as a script to my package.json, they run and error. I'll look at it a little more and see if I can isolate the issue. in the mean time like I said run the commands directly or use the watch versions will be fine.

Thanks!
Dave

**Comment by coryhouse** [comment]
What if you change the lint task in package.json to this?

`"lint": "esw webpack.config.* src tools",`

Or, does it work if you simplify it even further to this?

`"lint": "esw src",`

**Comment by digitalnoiz** [comment]
Alright.. I've done some googling and I'm less of a noob now.

The first true issue is that on windows I'm getting error results back from eslint-plugin-import that items are not resolved:

C:\dev\actdux\webpack.config.dev.js (1/0)
  ✖  1:21  Unable to resolve path to module 'webpack'  import/no-unresolved

C:\dev\actdux\src\api\mockAuthorApi.js (1/0)
  ✖  1:19  Unable to resolve path to module './delay'  import/no-unresolved

C:\dev\actdux\src\api\mockCourseApi.js (1/0)
  ✖  1:19  Unable to resolve path to module './delay'  import/no-unresolved

C:\dev\actdux\src\index.test.js (1/0)
  ✖  1:20  Unable to resolve path to module 'expect'  import/no-unresolved

C:\dev\actdux\tools\srcServer.js (4/0)
  ✖  1:21  Unable to resolve path to module 'express'                import/no-unresolved
  ✖  2:21  Unable to resolve path to module 'webpack'                import/no-unresolved
  ✖  4:20  Unable to resolve path to module '../webpack.config.dev'  import/no-unresolved
  ✖  5:18  Unable to resolve path to module 'open'                   import/no-unresolved

C:\dev\actdux\tools\startMessage.js (1/0)
  ✖  1:20  Unable to resolve path to module 'colors'  import/no-unresolved

✖ 9 errors (9:51:32 AM)

The npm error occurs when you run this as an npm script because eslint exits with an exit code of 1 because there are errors to report. That's indicated in the log above as the second entry for 14(?). To stop the npm error, you can run the command with the --silent flag and npm won't throw an error: `npm run lint --silent`.

Now I just need to sort out why eslint-plugin-import isn't happy.

**Comment by coryhouse** [comment]
These errors infer that some packages listed in package.json aren't installed on your machine. So either they're not installed or perhaps your [Node path is invalid](http://stackoverflow.com/a/32159233/26180). 

Check, the above Path. Then run `npm install` again. Do you receive any errors?

**Comment by digitalnoiz** [comment]
Well I got it working but I'm honestly a little baffled as to how.

I checked the path values and they were set as indicated in the stackoverflow article you linked. 

I noticed in the log that my npm version was 3.8.9 but the latest version is 3.9.3. It appears the node msi installer may not upgrade an existing npm install as that was the first thing I did before starting this project. I had to follow the upgrade instructions in their [documentation](https://www.npmjs.com/package/npm-windows-upgrade). After the upgrade was complete, linting still reported the no-resolve error.

Next I removed the node_modules directory from the project and ran `npm install` again. Linting still reported the no-resolve error.

This is where it get's weird. At that time I was working in a source directory under my users folder (c:\users\first.last\source) and on a whim I moved the project folder to a dev folder on the root of the drive (c:\dev). Linting stopped reporting an error and came back clean! Taken aback by the success, I then wondered if the issue had to do with the directory I was using so I moved it back to my source directory in my users folder. Linting reported no errors again! So I'm baffled but working so I call it a win.

**Comment by coryhouse** [comment]
Great! Sounds like it might have been a permissions issue with the initial directory. Once you did an install in a different directory it was already successfully installed, so you could move back to the former directory and run it fine.

Just one idea. Anyway, glad it's solved! :)

**Comment by coryhouse** [comment]
@digitalnoiz - I think I know why this failed for you. It was likely your path. Did the previous path you were using happen to have spaces in it? [eslint-watch doesn't support spaces in paths](https://github.com/coryhouse/react-slingshot/issues/150)

**Comment by digitalnoiz** [comment]
@coryhouse - I found some other references that running npm scripts in a directory that has a space in it's path will fail. It's interesting to note that running the commands directly work fine in a path with spaces, but making the same command a npm script then using "npm run..." does not. I haven't used the npm scripts before your course so I tested this further and saw how quickly it failed, so that's now on my checklist for node projects with npm scripts - no spaces in the project path. It was one of reasons that compelled me to try moving the directory closer to the root of my drive. But that's why my experience was weird. The path I was on initially didn't have any spaces and moving the directory back from the root directory to my users directory didn't re-introduce a failure as I expected. 

I tend to agree with you that it may have been and issue of permissions. My theory is that some packages directory permission didn't get set correctly when installed by npm, but the act of moving the directory caused windows to re-apply permissions to the sub directories. Then moving the project around would no longer encounter an issue as the permissions would have been fixed. I have no really good way to prove this but it fits the behavior. In any case, it's something to try now if I see node encountering errors resolving packages in the future.

Alternatively it may have been an issue of path resolution. As I indicated I updated npm in this process and there may have been the order of installing the dependencies first then updating npm that may have caused paths not to be resolved correctly for npm scripting. Because of that, I think it's rather important to point out that simply installing the latest node version from the MSI installer on a windows computer that had a previous node installation doesn't upgrade npm like it upgrades node. You have to follow npm's upgrade instructions in their [documentation](https://www.npmjs.com/package/npm-windows-upgrade) to get npm to update on windows.That seems a real big "gotcha" to me when your trying to keep things up to date as I was when I started the course.

I hope some of that helps if you see similar behavior with other students. Thanks for a great course. I finished it recently and I'm sold on the react / redux combination. The stack really makes sense to me and I hope to integrate it into larger projects soon. Thanks!

**Comment by bensezio** [comment]
I do  have a similar issue with  the lint

**Comment by vaishaliakhare** [comment]
when i run npm run lint i am getting below error

npm ERR! Darwin 15.3.0
npm ERR! argv "/Users/{userName}/.nvm/versions/node/v6.10.2/bin/node" "/Users/{userName}/.nvm/versions/node/v6.10.2/bin/npm" "run" "lint"
npm ERR! node v6.10.2
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE
npm ERR! development-environment@1.0.0 lint: `esw webpack.config.* src buildScripts --color`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the development-environment@1.0.0 lint script 'esw webpack.config.* src buildScripts --color'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the development-environment package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     esw webpack.config.* src buildScripts --color
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs development-environment
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls development-environment
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/{userName}/jssetup/js-dev-env/npm-debug.log

**Comment by coryhouse** [comment]
Hi @vaishaliakhare - Have you checked this list? https://github.com/coryhouse/pluralsight-redux-starter#having-issues-try-these-things-first

If that doesn't help, please check if this issue occurs on the final course exercise. If neither helps, please open a new issue. Thanks!

**Comment by vaishaliakhare** [comment]
Thanks @coryhouse .  This helped.

**Comment by plunk2000** [comment]
What was the output of the original script? Did it find any linting errors?

If you have linting errors, eslint returns 1 as the exit code.
